src/mpyl/steps/deploy/dagster.py

```diff
@@ -0,0 +1,131 @@
+            docker_config=DockerConfig.from_dict(step_input.run_properties.config),
+        )
+        user_code_name_to_deploy = user_code_deployment["deployments"][0]["name"]
+
+        self._logger.debug(f"Deploying user code with values: {user_code_deployment}")
+
+        deploy_result = helm.install_with_values_yaml(
+            logger=self._logger,
+            step_input=step_input,
+            values=user_code_deployment,
+            release_name=convert_name_to_helm_release_name(
+                step_input.project.name, name_suffix
+            ),
+            chart_name="dagster/dagster-user-deployments",
+            namespace=namespace,
+            kube_context=context,
+        )
+
+        if deploy_result.success and not step_input.dry_run:
+            config_map = get_config_map(context, namespace, "dagster-workspace-yaml")
+            dagster_workspace = yaml.safe_load(config_map.data["workspace.yaml"])
+
+            server_names = [
+                w["grpc_server"]["location_name"]
+                for w in dagster_workspace["load_from"]
+            ]
+
+            # If the server new (not in existing workspace.yml), we append it
+            if user_code_name_to_deploy not in server_names:
+                self._logger.info(
+                    f"Adding new server {user_code_name_to_deploy} to dagster's workspace.yaml"
+                )
+                dagster_workspace["load_from"].append(
+                    to_grpc_server_entry(
+                        host=user_code_deployment["deployments"][0]["name"],
+                        port=user_code_deployment["deployments"][0]["port"],
+                        location_name=user_code_deployment["deployments"][0]["name"],
+                    )
+                )
+                updated_config_map = update_config_map_field(
+                    config_map=config_map,
+                    field="workspace.yaml",
+                    data=dagster_workspace,
+                )
+                replaced_configmap_result = replace_config_map(
+                    self._logger,
+                    context,
+                    "dagster",
+                    "dagster-workspace-yaml",
+                    updated_config_map,
+                )
+                if not replaced_configmap_result.success:
+                    return replaced_configmap_result
+
+            # restarting ui and daemon
+            rollout_restart_output = rollout_restart_deployment(
+                self._logger, namespace, "dagster-dagit"
+            )
+            if rollout_restart_output.success:
+                rollout_restart_output = rollout_restart_deployment(
+                    self._logger, namespace, "dagster-dagit"
+                )
+                if not rollout_restart_output.success:
+                    return rollout_restart_output
+            else:
+                return rollout_restart_output
+        return deploy_result
```

Comment:
use `output` consistently

Reply:
Hmm, the benefit of this approach instead of just having a try catch block around this entire `if deploy_result.success and not step_input.dry_run` code block is that you have more precise error messages on failures. But it does make the code a bit messy, not sure what's a better approach actually.